python/lsst/daf/butler/core/datasets/ref.py

```diff
@@ -114,9 +114,8 @@ def direct(
         return node
 
 
-DatasetId = Union[int, uuid.UUID]
-"""A type-annotation alias for dataset ID which could be either integer or
-UUID.
+DatasetId = uuid.UUID
+"""A type-annotation alias for dataset ID providing typing flexibility.
```

Comment:
Someday I'd like to get rid of this type alias entirely, as I'm pretty confident we'll always use UUID and `DatasetId` is too similar to `DataId` for my comfort.  But not on this ticket.